[18.0][MIG] sale_order_priority: Migration to 18.0

apami0111 opened the pull request:
- Migrated from 17.0 to 18.0.
- Updated xpath expression in the view to reflect Odoo 18.0 changes (//list instead of //tree).
- Test class now uses Command.create to define order_line values, in line with Odoo 18.0 conventions.

SimoRubi committed:
[ADD] sale_order_priority (#657)

* [ADD] sale_order_priority

* Code improvements

* [FIX] Better not to call write in inverse

Commit:
[MIG] Migrate sale_order_priority to 12.0

MiquelRForgeFlow committed:
[IMP] sale_order_priority: black, isort, prettier

MiquelRForgeFlow committed:
[MIG] sale_order_priority: Migration to 13.0

GuillemCForgeFlow committed:
[14.0][IMP]sale_order_priority: black, isort, prettier

GuillemCForgeFlow committed:
[14.0][MIG]sale_order_priority: Migration to 14.0

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_order_priority
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_order_priority/it/

Commit:
[MIG] sale_order_priority: Migration to 15.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-15.0/sale-workflow-15.0-sale_order_priority
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-15-0/sale-workflow-15-0-sale_order_priority/

Ivorra78 committed:
Added translation using Weblate (Spanish)

astirpe committed:
[MIG] sale_order_priority: pre-commit

astirpe committed:
[MIG] sale_order_priority: migrate to V17

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (7 of 7 strings)

Translation: sale-workflow-17.0/sale-workflow-17.0-sale_order_priority
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-17-0/sale-workflow-17-0-sale_order_priority/it/

Commit:
[FIX]sale_order_priority: Fix tests

apami0111 committed:
[MIG] sale_order_priority: Migration to 18.0

rousseldenis commented:
/ocabot migration sale_order_priority

rousseldenis commented on sale_order_priority/__manifest__.py:
```diff
@@ -3,7 +3,7 @@
     "name": "Sale Order Priority",
     "summary": "Define priority on sale orders",
-    "version": "17.0.1.0.1",
+    "version": "18.0.1.0.1",
```
Please reset to `18.0.1.0.0`

rousseldenis commented on sale_order_priority/models/sale.py:
```diff
@@ -0,0 +1,47 @@
+                line.priority = priority
+
+    def action_confirm(self):
+        return super(
```
This is not correct as `self` can be multi recordset.

Moreover, I think the transmission of the priority should be done through procurement if possible as context should be avoided.

apami0111 commented:
Thank you very much for your comment. I will be working on the suggested
changes and will update the PR accordingly.

El vie, 4 jul 2025 a las 9:54, Denis Roussel (ACSONE) (<
***@***.***>) escribió:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In sale_order_priority/__manifest__.py
> <https://github.com/OCA/sale-workflow/pull/3786#discussion_r2185545599>:
>
> > @@ -3,7 +3,7 @@
>  {
>      "name": "Sale Order Priority",
>      "summary": "Define priority on sale orders",
> -    "version": "17.0.1.0.1",
> +    "version": "18.0.1.0.1",
>
> Please reset to 18.0.1.0.0
> ------------------------------
>
> In sale_order_priority/models/sale.py
> <https://github.com/OCA/sale-workflow/pull/3786#discussion_r2185572856>:
>
> > +    @api.depends("order_line.priority")
> +    def _compute_priority(self):
> +        for order in self.filtered(lambda x: x.order_line):
> +            priority = order.mapped("order_line.priority")
> +            order.priority = max([x for x in priority if x] or "0")
> +
> +    def _inverse_priority(self):
> +        for order in self:
> +            priority = order.priority
> +            for line in order.order_line.filtered(
> +                lambda x, pr=priority: x.priority != pr
> +            ):
> +                line.priority = priority
> +
> +    def action_confirm(self):
> +        return super(
>
> This is not correct as self can be multi recordset.
>
> Moreover, I think the transmission of the priority should be done through
> procurement if possible as context should be avoided.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/sale-workflow/pull/3786#pullrequestreview-2987512902>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BKVNRJ3QUYCLX6VP2YT44433G2IT3AVCNFSM6AAAAACAUN623OVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDSOBXGUYTEOJQGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

apami0111 commented:
@rousseldenis Thanks for your feedback! I've addressed the issue with the multi-recordset by adjusting the logic accordingly. I’ve also reworked the way priority is transmitted, avoiding the use of context as suggested.

The following structural changes were made as well:
- Renamed: sale_order_priority/models/{sale.py => sale_order.py}
- Created: sale_order_priority/models/sale_order_line.py
- Created: sale_order_priority/models/stock_move.py
- Removed: sale_order_priority/models/stock_picking.py

Feel free to review the updated implementation. Let me know if anything else needs to be improved.

celm1990 reviewed (approved):
LGTM

marcelsavegnago reviewed (approved):
LGTM

BhaveshHeliconia reviewed (approved):
Functional review LGTM!

rousseldenis commented:
/ocabot merge nobump
